File: frontend/lib/threat-intel.ts
import { apiRequest } from "./api";

export type IndicatorType = "ip" | "domain" | "hash" | "cve";

export type ThreatIntelResult = {
  indicator: string;
  type: IndicatorType;
  risk_score: number;
  reputation: string;
  tags: string[];
  recommendations: string[];
  severity?: string | null;
  cvss?: number | null;
  description?: string | null;
  mitigation: string[];
};

export type ThreatIntelPayload = {
  indicator: string;
  type: IndicatorType;
};

export async function enrichThreatIndicator(payload: ThreatIntelPayload, token: string) {
  return apiRequest<ThreatIntelResult>(
    "/api/threat-intel/enrich",
    {
      method: "POST",
      body: JSON.stringify(payload),
    },
    token,
  );
}

export async function fetchIpIntel(ip: string, token: string) {
  return apiRequest<ThreatIntelResult>(`/api/threat-intel/ip/${encodeURIComponent(ip)}`, { method: "GET" }, token);
}

export async function fetchCveIntel(cve: string, token: string) {
  return apiRequest<ThreatIntelResult>(`/api/threat-intel/cve/${encodeURIComponent(cve)}`, { method: "GET" }, token);
}

File: backend/app/schemas/threat_intel.py
from typing import Literal

from pydantic import BaseModel, Field

IndicatorType = Literal["ip", "domain", "hash", "cve"]


class ThreatIntelRequest(BaseModel):
    indicator: str = Field(..., min_length=2, max_length=255)
    type: IndicatorType


class ThreatIntelResult(BaseModel):
    indicator: str
    type: IndicatorType
    risk_score: int = Field(..., ge=0, le=100)
    reputation: str
    tags: list[str]
    recommendations: list[str]
    severity: str | None = None
    cvss: float | None = None
    description: str | None = None
    mitigation: list[str] = []
